mlir/utils/jenkins/Jenkinsfile

```diff
@@ -170,180 +288,118 @@ pipeline {
-                            rebootNode()
+                    stage("Shared Library: random E2E tests") {
+                        when {
+                            beforeAgent true;
+                            allOf {
+                                equals expected: true, actual: params.sharedLib;
+                                equals expected: true, actual: params.nightly
+                            }
                         }
-                        always {
-                            cleanWs()
+                        steps {
+                            check_randomE2ETests()
                         }
                     }
-                }
-                stage("Static Library PR tests") {
-                    when {
-                        beforeAgent true;
-                        allOf {
+                    stage("Static Library: test librockCompiler with MIOpen") {
+                        when {
+                            beforeAgent true;
                             equals expected: true, actual: params.staticLib;
                             equals expected: false, actual: params.nightly;
-                            equals expected: false, actual: params.weekly;
                         }
-                    }
-                    agent {
-                        docker {
-                            image dockerImage()
-                            args dockerArgs()
-                            label "mlir && ${params.canXdlops ? '(gfx908 || gfx90a)' : 'gfx906' }"
-                            alwaysPull true
+                        environment {
+                            // Make libMIOpen.so accessible to the test driver
+                            LD_LIBRARY_PATH="${WORKSPACE}/MIOpen/build/lib:$LD_LIBRARY_PATH:"
                         }
-                    }
-                    environment {
-                        PATH="/opt/rocm/llvm/bin:$PATH"
-                        HOME="${WORKSPACE}"
-                    }
-                    stages {
-                        stage('Environment') {
-                            steps {
-                                showEnv()
-                            }
+                        steps {
+                            sh 'rm build/CMakeCache.txt'
+                            buildMIOpenWithMLIR()
+                            testMIOpenDriver(/*testInt8=*/true, /*tuning=*/false, "selected-resnet50-miopen-configs")
+                            testMIOpenDriver(/*testInt8=*/true, /*tuning=*/true, "selected-resnet50-miopen-configs")
                         }
-                        stage("Build MIOpen with librockCompiler") {
-                            steps {
-                                buildMIOpenWithMLIR()
+                    }
+                    stage('Static Tests') {
+                        when {
+                            allOf {
+                                equals expected: false, actual: params.nightly;
```

Comment:
One bit of weirdness that could be possible: the static checks assume you have a compile_commands.json around, and so need the project to be configured - so if we try to cheat the CI (as we sometimes need to do) this'll fall over without any way to fix it. Since we do want these tests to run always, how about this:
- Split the shared library build and test into a build step (`check-mlir-build-only check-rocmlir-build-only`) and a test step (`check-mlir` `check-rocmlir`) and stick the static tests in between. You could even have them in the same `steps` block - I think Jenkins has a `when{}` at that level.